### DISCUSSION TITLE
WalletConnect signer support

### DISCUSSION THREAD

**Discussion by rossbulat** [polkadot-api/polkadot-api#871] [Q&A] [answered]
Using WalletConnect derived signatures via `getPolkadotSignerFromPjs` currently yields a `Bad Proof` error when submitting a transaction. My implementation is using `signSubmitAndWatch`.

I imagine there is a payload discrepancy somewhere along the line, but not my expertise to debug the internals of this. 

```
signer = getPolkadotSignerFromPjs(
   address,
   signWcTx, // <-- sends wallet connect request and returns { signature } 
   // Signing bytes not currently being used.
      async () => ({
         id: 0,
        signature: '0x',
       })
   );
```

`signWcTx` submits a request in the same fashion as https://docs.reown.com/advanced/multichain/polkadot/dapp-integration-guide#polkadot-example, and returning a signature hex string.

For Staking this seems to be the only outstanding action item before the PAPI version is on parity with the PJS version. I'd be interested to what the Polkadot API team think is the best way forward here, maybe another custom signer for Wallet Connect specifically will be useful, depending on what is causing this Bad Proof error.

Thanks for considering, and please let me know if I can give further context.

**Comment by voliva** [comment, marked as answer]
In theory you are on the right track, on first sight this should work.

I did implement a rough version of a WalletConnect signer for the papi-console, and the happy path of signing a System.remark on Westend with Subwallet seems to be working fine. It's using `getPolkadotSignerFromPjs` + the dapp integration guide from reown through `client.request`: https://github.com/polkadot-api/papi-console/blob/6640a09adc68dd91bbab9471066acda193ae9842/src/state/walletconnect.state.ts#L187.

This code is extremely rough, I added it as an initial setup to have multiple options for signing transactions, not only from pjs extensions, and to have it as a small example. Once it's more fleshed out, I think it would make sense to move it as a specific walletConnect signer on the main repo.

Not really sure what could be happening in your case, or if I overlooked something and my implementation only works on the specific configuration I'm using. Can you provide a reproduction case? Or give more details on `signWcTx`, what's it doing differently from the example?

**Reply by rossbulat** [reply]
Super helpful, thank you @voliva! My `signWcTx` function was destructuring `signature` and returning that, instead of returning the whole request result. Your example solved this issue, thank you 🙏.

This can be closed now, unless we take things forward and you opt to create a specific walletConnect signer.
